### DISCUSSION TITLE
Select Virtual List

### DISCUSSION THREAD

**Discussion by hzh11012** [cosscom/coss#155] [Suggestions]
Hope Select can enable virtual scrolling for big data

**Comment by pasqualevitiello** [comment]
Isn't it already? 😊


<img width="422" alt="Screenshot 2024-12-17 at 12 43 50" src="https://github.com/user-attachments/assets/3cbd2903-6202-4e6e-93f0-9b7ae1b227c4" />

**Reply by gregmsanderson** [reply]
I don't think it is. I think _all_ items are added to the DOM. That makes it slow to render a huge list.

Since I've just been looking at this too, I'd guess the request is for an equivalent to the `isVirtualized` prop in NextUI. They use React Aria hooks rather than Radix however they use [@tanstack/react-virtual](https://tanstack.com/virtual/latest) to add virtualization to their Select: https://nextui.org/docs/components/select#virtualization

That way even if your list has 10,000 items it still loads quickly. Only the ones in view are included.

I tried adding it however it seemed to cause issues with Radix (since its portal works differently to how Aria does it).

**Comment by pasqualevitiello** [comment]
Misread the initial message. For reference: https://github.com/shadcn-ui/ui/issues/543 there should be a couple of working examples - I hadn't tested these personally.
